lib/bounces.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const error = require('./error')
+const P = require('./promise')
+
+module.exports = (config, db) => {
+  const configBounces = config.smtp && config.smtp.bounces || {}
+  const BOUNCES_ENABLED = !! configBounces.enabled
+  const MAX_HARD = configBounces.hard && configBounces.hard.max || 0
+  const MAX_SOFT = configBounces.soft && configBounces.soft.max || 0
+  const MAX_COMPLAINT = configBounces.complaint && configBounces.complaint.max || 0
+  const DURATION_HARD = configBounces.hard && configBounces.hard.duration || Infinity
+  const DURATION_SOFT = configBounces.soft && configBounces.soft.duration || Infinity
+  const DURATION_COMPLAINT = configBounces.complaint && configBounces.complaint.duration || Infinity
+  const BOUNCE_TYPE_HARD = 1
+  const BOUNCE_TYPE_SOFT = 2
+  const BOUNCE_TYPE_COMPLAINT = 3
+
+  const freeze = Object.freeze
+  const BOUNCE_RULES = freeze({
+    [BOUNCE_TYPE_HARD]: freeze({
```

Comment:
> Looks like it is new in ES6, you have the ability to compute a key.

Oh, so bona fide language feature rather than weird cleverness. It reads like some ghastly overloading of array literal syntax to me, but I'm sure I'll get over that.